src/LegalizeVectors.cpp

```diff
@@ -429,8 +429,8 @@ class LegalizeVectors : public IRMutator {
     Expr visit(const Shuffle *op) override {
         internal_assert(op->type.lanes() <= max_lanes) << Expr(op);
         bool requires_mutation = false;
-        for (size_t i = 0; i < op->vectors.size(); ++i) {
-            if (op->vectors[i].type().lanes() > max_lanes) {
+        for (auto &&vec : op->vectors) {
+            if (vec.type().lanes() > max_lanes) {
```

Comment:
```suggestion
        for (const auto &vec : op->vectors) {
            if (vec.type().lanes() > max_lanes) {
```

Does this not satisfy clang-tidy? Or even this?

```suggestion
        for (const Expr &vec : op->vectors) {
            if (vec.type().lanes() > max_lanes) {
```

Reply:
Yeah it would. This was not the case I care about in particular. I was reading about what the use is of `auto&&`  and why clang-tidy sometimes suggests with one & and sometimes with &&. Turns out that iterating over a std::vector<bool> requires auto&& if you wish to adjust the value of the stored bool. The reason being that vector is specialized to store bools as one bit, meaning that the iterator gives you a wrapper struct `std::vector<bool>::reference` with overloaded assignment operators to be able to overwrite the bit. So `for (bool &val : vector_of_bools)` doesn't compile: you have to auto&&, because for some weird reason which I didn't understand, also `auto&` wouldn't compile: https://godbolt.org/z/4os7TqMWo